source/common/upstream/cluster_manager_impl.cc

```diff
@@ -715,8 +718,29 @@ Host::CreateConnectionData ClusterManagerImpl::tcpConnForCluster(const std::stri
 
   HostConstSharedPtr logical_host = entry->second->lb_->chooseHost(context);
   if (logical_host) {
-    auto conn_info =
-        logical_host->createConnection(cluster_manager.thread_local_dispatcher_, nullptr);
+
```

Comment:
I don't think you should put this logic in ClusterManager, they should live in tcp_proxy.

Reply:
@lizan I see, let me refactor the code, thanks!